src/thing_shots.c

```diff
@@ -476,16 +477,21 @@ TbBool shot_hit_wall_at(struct Thing *shotng, struct Coord3d *pos)
                 apply_damage_to_thing(doortng, i, shotst->damage_type, -1);
             } else
             {
-                struct SlabMap* slb = get_slabmap_for_subtile(stl_num_decode_x(hit_stl_num), stl_num_decode_y(hit_stl_num));
-                if ( ( (old_health <= slb->health) && (slb->kind != SlbT_GEMS) ) || (!digging) )
+                short eff_kind = shotst->hit_generic.effect_model;
+                short smpl_idx = shotst->hit_generic.sndsample_idx;
+                unsigned char range = shotst->hit_generic.sndsample_range;
+                if (digging)
                 {
-                    efftng = create_shot_hit_effect(&shotng->mappos, shotng->owner, shotst->hit_generic.effect_model, shotst->hit_generic.sndsample_idx, shotst->hit_generic.sndsample_range);
-                }
-                else
-                {
-                    struct Thing *diggertng = thing_get(shotng->parent_idx);
-                    efftng = create_shot_hit_effect(&diggertng->mappos, diggertng->owner, shotst->dig.effect_model, shotst->dig.sndsample_idx, shotst->dig.sndsample_range);
+                    struct SlabMap* slb = get_slabmap_for_subtile(stl_num_decode_x(hit_stl_num), stl_num_decode_y(hit_stl_num));
+                    if ((old_health > slb->health) || (slb->kind == SlbT_GEMS))
+                    {
+                        smpl_idx = shotst->dig.sndsample_idx;
+                        range = shotst->dig.sndsample_range;
+                        eff_kind = shotst->dig.effect_model;
+                    }
                 }
+                efftng = create_shot_hit_effect(&shotng->mappos, shotng->owner, eff_kind, smpl_idx, range);
```

Comment:
Won't the rubble go into the wall if we use shotng->mappos?

Reply:
It looks very nice, coming from where you hit the wall instead of from where your feet are.

Reply:
But I noticed that the rocks sometimes go into the wall, which we don't want, surely?

Reply:
It would be nicer if the rocks fell, but the bigger rocks from destroying dirt also get stuck in neighboring dirt slabs. It seems more correct to me then rocks spawning at the feet, with further improvement possible to make effect elements not stay stuck in walls, but that's for blood splats and everything too.

I still prefer how this looks. Do you not share this view?

Reply:
In instf_dig, the rocks spawn at the digger's location, though that's probably because it can't spawn anywhere else as there's technically no shot.

Reply:
yes, and you can't really notice it anyway, since you're watching from above.